python/ray/data/impl/arrow_block.py

```diff
@@ -399,3 +397,79 @@ def _copy_table(table: "pyarrow.Table") -> "pyarrow.Table":
             arr = col.combine_chunks()
         new_cols.append(arr)
     return pa.Table.from_arrays(new_cols, schema=table.schema)
+
+
+def _arrow_array_reduce(a: "pyarrow.Array"):
+    """Custom reducer for Arrow arrays that works around a zero-copy slicing pickling
+    bug.
+
+    See https://issues.apache.org/jira/browse/ARROW-10739.
+    """
+    # TODO(Clark): Do we need to generically support children arrays?
+    import pyarrow as pa
+    from ray.data.extensions import ArrowTensorArray, ArrowTensorType
+
+    if isinstance(a, ArrowTensorArray):
```

Comment:
Could this just be defined directly in our ArrowTensorArray class?

Reply:
Yep I have a soft TODO for consolidating a lot of this logic and the logic in `ArrowTensorArray._to_numpy()` into a set of shared utilities on the `ArrowTensorArray` class!